# APM111 - Statistical Theory - FA 6

In [27]:
import pandas as pd
import numpy as np

## 1. Table 1 shows a frequency distribution of grades on a final examination in college algebra. Find the quartiles of the distribution.

In [28]:
df = pd.read_csv('algebra.csv')
df

Grade  Number of Students
0  90-100                   9
1   80-89                  32
2   70-79                  43
3   60-69                  21
4   50-59                  11
5   40-49                   3
6   30-39                   1

In [29]:
def expand_grades(row):
    range_start, range_end = map(int, row['Grade'].split('-'))
    return [value for value in range(range_start, range_end + 1) for _ in range(row['Number of Students'])]

expanded_grades = df.apply(expand_grades, axis=1).explode().astype(int)

### Quartiles

In [30]:
# Calculate quartiles
Q1 = expanded_grades.quantile(0.25)
Q2 = expanded_grades.quantile(0.5)
Q3 = expanded_grades.quantile(0.75)

print(f'''Quartiles 
25th Percentile: {Q1}
50th Percentile: {Q2}
75th Percentile: {Q3}''')

Quartiles 
25th Percentile: 67.0
50th Percentile: 75.0
75th Percentile: 83.0


## 2a. On a final examination in statistics, the mean grade of a group of 150 students was 78 and the standard deviation was 8.0. In algebra, however, the mean final grade of the group was 73 and the standard deviation was 7.6. In which subject was there the greater absolute dispersion

In [34]:
number_of_students = 150

# Statistics
stats_mean = 78
stats_sd = 8

# Algebra
algebra_mean = 73
algebra_sd = 7.6

if stats_sd > algebra_sd:
    print(f'The statistics subject has greater absolute dispersion, as indicated by its higher standard deviation ({stats_sd} vs. {algebra_sd}).')
elif stats_sd < algebra_sd:
    print(f'The algebra subject has greater absolute dispersion, as indicated by its higher standard deviation ({algebra_sd} vs. {stats_sd}).')
else:
    print(f'Neither of the subject has the greater absolute dispersion due to both having the same standard deviation.')

The statistics subject has greater absolute dispersion, as indicated by its higher standard deviation (8 vs. 7.6).


## 2b. In which subject was there the greater (b) relative dispersion?

In [36]:
# Statistics
stats_cv = round(stats_sd/stats_mean, 4)

# Algebra
algebra_cv = round(algebra_sd/algebra_mean, 4)

if stats_cv > algebra_cv:
    print(f'The statistics subject has greater relative dispersion, as indicated by its higher relative dispersion ({stats_cv} vs {algebra_cv}).')
elif stats_cv < algebra_cv:
    print(f'The algebra subject has greater relative dispersion, as indicated by its higher relative dispersion ({algebra_cv} vs {stats_cv}).')
else:
    print(f'Neither of the subject has the greater relative dispersion due to both having the same value of relative dispersion.')

The algebra subject has greater relative dispersion, as indicated by its higher relative dispersion (0.1041 vs 0.1026).


## 3. Prove that the mean and standard deviation of a set of standard scores are equal to 0 and 1, respectively. Use the following problem to illustrate this: Convert the set 6, 2, 8, 7, 5 into standard scores.

### a. Calculate the Mean (𝜇) of the Original Data Set

In [37]:
data_set = [6, 2, 8, 7, 5]

In [39]:
data_set_mean = np.mean(data_set)
print(f'Data set mean: {data_set_mean}')

Data set mean: 5.6


$$
μ = \frac{6+2+8+7+5}{5} = \frac{28}{5} = 5.6
$$

### b. Calculate the Standard Deviation (𝜎) of the Original Data Set

In [59]:
# Calculate Each Deviation from the Mean
dev_set = []
for i in range(len(data_set)):
    dev = round(data_set[i] - data_set_mean, 1)
    dev_set.append(dev)

print(f'Deviation: {dev_set}')

# Square Each Deviation
dev_squared = []
for i in range(len(dev_set)):
    square = round(dev_set[i] ** 2, 2)
    dev_squared.append(square)
    
print(f'Square Deviation: {dev_squared}')
    
# Calculate the Variance
var = sum(dev_squared)/len(dev_squared)

print(f'Variance: {var}')

# Calculate the Standard Deviation
sd = round(np.sqrt(var), 2)

print(f'Standard Deviation: {sd}')

Deviation: [0.4, -3.6, 2.4, 1.4, -0.6]
Square Deviation: [0.16, 12.96, 5.76, 1.96, 0.36]
Variance: 4.24
Standard Deviation: 2.06


#### Calculate Each Deviation from the Mean:
$$
6 - 5.6 = 0.4 \\
2 - 5.6 = -3.6 \\
8 - 5.6 = 2.4 \\
7 - 5.6 = 1.4 \\
5 - 5.6 = -0.6
$$

#### Square Each Deviation:
$$
0.4^2 = 0.16 \\
-3.6^2 = 12.96 \\
2.4^2 = 5.76 \\
1.4^2 = 1.96 \\
-0.6^2 = 0.36
$$

#### Calculate the Variance:
$$
Variance = \frac{0.16+12.96+5.76+1.96+0.36}{5} = \frac{21.2}{5} = 4.24
$$

#### Calculate the Standard Deviation:
$$
σ = \sqrt{4.24} ≈ 2.06
$$

### c. Convert the Data Set into Standard Scores (z-scores)

In [62]:
z_scores = []
for i in range(len(data_set)):
    z_score = round((data_set[i] - data_set_mean)/sd, 3)
    z_scores.append(z_score)
    
z_scores

[0.194, -1.748, 1.165, 0.68, -0.291]

Using the formula:
$$

$$